lib/beaker-hostgenerator/data.rb

```diff
@@ -295,6 +295,20 @@ def osinfo
             ]
           }
         },
+        'centos9-64' => {
+          :general => {
+            # There are no EL9 builds just yet - pretend it's EL8
+            'platform' => 'el-8-x86_64'
```

Comment:
I tried setting `platform` to `el-9-x86_64` and `packaging_platform` to `el-8-x86_64` but beaker-puppet only looks at `platform` so I don't really know what the point of `packaging_platform`. I found some references for Puppet 5 for example, but it looks like it's largely irrelevant.

Reply:
And the EL8 packages fail on this:
```
centos9-64-1 17:30:38$ yum -y  install puppet-agent
  Puppet 7 Repository el 8 - x86_64               9.0 MB/s | 8.1 MB     00:00    
  Last metadata expiration check: 0:00:01 ago on Thu Sep  9 15:30:40 2021.
  Error: 
   Problem: cannot install the best candidate for the job
    - nothing provides /bin/mkdir needed by puppet-agent-7.10.0-1.el8.x86_64
  (try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

Reply:
lol. no mkdir? is it in a different path or really not present in the image?

Reply:
In EL9 it's `/usr/bin/mkdir` so the EL8 packages (which has `/bin/mkdir`) doesn't work.

Reply:
> I believe `packaging_platform` is/was used for platforms on which puppetlabs packages are built (i.e. all windows packages are built on 2012, and aix packages are build on 7.1). On the FOSS side I don't see this being used anymore, but there's still some logic depending on this in our PE tooling.

I recently had a look at beaker-puppet and I actually want to start using it again where possible. IMHO it still makes sense and consider it a bug that it's ignored.

> I feel it's redundant in cases where `platform` and `packaging_platform` are identical, so I would remove `packaging_platform` in those cases, but I would keep it for platforms on which the 2 differ.

Agreed. No need to duplicate when they match.